libtiledbvcf/src/stats/variant_stats_reader.cc

```diff
@@ -278,11 +521,16 @@ void VariantStatsReader::compute_af_worker_() {
 
   // Setup the query
   ManagedQuery mq(array_, "variant_stats", TILEDB_UNORDERED);
-  mq.select_columns({"pos", "allele", "ac"});
+  mq.select_columns({"pos", "sample", "allele", "ac", "an", "end"});
   mq.select_point<std::string>("contig", regions_.front().seq_name);
   for (auto& region : regions_) {
     LOG_DEBUG("[VariantStatsReader] compute_af for region={}", region.to_str());
-    mq.select_ranges<uint32_t>("pos", {{region.min, region.max}});
+    mq.select_ranges<uint32_t>(
+        "pos",
+        {{region.min - ((variant_stats_version > 2) ?
```

Comment:
This logic is asking for an intermediate variable.

Reply:
This change extends the start to include relevant ref blocks. It does so only for variant stats ≥ 3, and doesn't assign a start position less than the beginning of the contig.